src/plots/stackedAreaPlot.ts

```diff
@@ -202,6 +181,33 @@ export module Plots {
       return { x: pixelPoint.x, y: scaledYValue };
     }
 
+    protected _getDataToDraw() {
+      if (super.downsamplingEnabled() === false) {
+        return super._getDataToDraw();
+      }
+      let dataToDraw = new Utils.Map<Dataset, any[]> ();
+      if (this.datasets.length === 0) {
+        return dataToDraw;
+      }
+      let overallfilteredDataIndices: number[] = [];
+      this.datasets().forEach((dataset, i) => {
+        let data = dataset.data();
+        let filteredDataIndices = data.map((d, i) => i);
+        if (super.croppedRenderingEnabled()) {
+          filteredDataIndices = super._filterCroppedRendering(dataset, filteredDataIndices);
+        }
+        if (super.downsamplingEnabled()) {
+          filteredDataIndices = super._filterDownsampling(dataset, filteredDataIndices);
+        }
+        overallfilteredDataIndices = overallfilteredDataIndices.concat(filteredDataIndices
+          .filter((index) => {return overallfilteredDataIndices.indexOf(index) < 0; }));
+      });
+      this.datasets().forEach((dataset, i) => {
+        let data = dataset.data();
+        dataToDraw.set(dataset, [overallfilteredDataIndices.sort((a, b) => a - b).map((d, i) => data[d])]);
```

Comment:
We want the indices to be sorted so that the result dataset of mapping function keep the same order as the original dataset.